packages/features/src/common/store/vault.test.ts

```diff
@@ -70,43 +70,84 @@ describe('VaultStore', () => {
   })
 
   it('returns current wallet', () => {
-    const { result } = renderHook(() => useVaultStore())
+    addCredential()
     act(() => {
-      result.current.addCredential({
-        walletPublicKey: 'pubkey',
-        walletPrivateKey: 'privkey',
-        walletName: 'Test Wallet'
-      })
       result.current.setCurrentWalletPublicKey({
-        currentWalletPublicKey: 'pubkey'
+        currentWalletPublicKey: credential.walletPublicKey
       })
     })
-    expect(result.current.getCurrentWallet()?.walletPublicKey).toEqual('pubkey')
+    expect(result.current.getCurrentWallet()?.walletPublicKey).toEqual(
+      credential.walletPublicKey
+    )
   })
 
-  it('creates a new wallet', async () => {
-    const walletName = 'TestWallet'
-    const { result } = renderHook(() => useVaultStore())
+  const performWalletTests = async (walletName: string, network: Network) => {
+    const accountNumber = 0
+    resetVault()
     await act(async () => {
-      const { mnemonic } = await result.current.createWallet({ walletName })
-      expect(mnemonic.length).gt(0)
+      const wallet = await result.current.createWallet({
+        walletName,
+        network,
+        accountNumber
+      })
+      expect(wallet.mnemonic.length).gt(0)
     })
     expect(result.current.credentials.length).toEqual(1)
     expect(result.current.getCurrentWallet()?.walletName).toEqual(walletName)
+  }
+
+  it('creates a new wallet for Mina networks', async () => {
+    await performWalletTests('TestWallet', Network.Mina)
   })
 
-  it('restores wallet', async () => {
-    const walletName = 'NiceWallet'
+  it('restores wallet for Mina networks', async () => {
```

Comment:
I think this `it` block is doubled accidentally 😄

Reply:
Oh! whoops 😅